private/bufpkg/bufapimodule/module_reader_test.go

```diff
@@ -22,18 +22,95 @@ import (
 
 	"github.com/bufbuild/buf/private/bufpkg/bufmodule/bufmoduleref"
 	"github.com/bufbuild/buf/private/gen/proto/connect/buf/alpha/registry/v1alpha1/registryv1alpha1connect"
-	v1alpha1 "github.com/bufbuild/buf/private/gen/proto/go/buf/alpha/module/v1alpha1"
+	modulev1alpha1 "github.com/bufbuild/buf/private/gen/proto/go/buf/alpha/module/v1alpha1"
 	registryv1alpha1 "github.com/bufbuild/buf/private/gen/proto/go/buf/alpha/registry/v1alpha1"
+	"github.com/bufbuild/buf/private/pkg/manifest"
+	"github.com/bufbuild/buf/private/pkg/storage/storagemem"
 	"github.com/bufbuild/connect-go"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 )
 
 type mockDownloadService struct {
-	module   *v1alpha1.Module
-	manifest *v1alpha1.Blob
-	blobs    []*v1alpha1.Blob
-	err      error
+	module       *modulev1alpha1.Module
+	manifestBlob *modulev1alpha1.Blob
+	blobs        []*modulev1alpha1.Blob
+	err          error
+}
+
+type option interface {
+	apply(*mockDownloadService) error
+}
+
+type filemap map[string][]byte
+
+func (fm filemap) apply(m *mockDownloadService) error {
+	bucket, err := storagemem.NewReadBucket(fm)
+	if err != nil {
+		return err
+	}
+	ctx := context.Background()
+	moduleManifest, blobSet, err := manifest.NewFromBucket(ctx, bucket)
+	if err != nil {
+		return err
+	}
+	mBlob, err := moduleManifest.Blob()
+	if err != nil {
+		return err
+	}
+	m.manifestBlob, err = manifest.AsProtoBlob(ctx, mBlob)
+	if err != nil {
+		return err
+	}
+	blobs := blobSet.Blobs()
+	m.blobs = make([]*modulev1alpha1.Blob, 0, len(blobs))
+	for _, blob := range blobs {
+		protoBlob, err := manifest.AsProtoBlob(ctx, blob)
+		if err != nil {
+			return err
+		}
+		m.blobs = append(m.blobs, protoBlob)
+	}
+	return nil
+}
+
+func WithBlobsFromMap(files map[string][]byte) option {
```

Comment:
Nope, just was a natural typing of an option func. It's a test package, so it can't be included anyway. Still, I'll lowcase it and move it to the bottom of the file per style.